keps/sig-storage/1472-storage-capacity-tracking/README.md

```diff
@@ -281,63 +280,89 @@ still an unsolved problem.
-
-    // Status contains the properties that can change over time.
-    Status CSIStorageCapacityStatus
-}
-
-// CSIStorageCapacitySpec contains the fixed properties of one capacity value.
-// one capacity value.
-type CSIStorageCapacitySpec struct {
-    // The CSI driver that provides the storage.
-    // This must be the string returned by the CSI GetPluginName() call.
-    DriverName string
-
-    // NodeTopology defines which nodes have access to the storage for which
-    // capacity was reported. If not set, the storage is accessible from all
-    // nodes in the cluster.
-    // +optional
-    NodeTopology *v1.NodeSelector
-
-    // The storage class name of the StorageClass which provided the
-    // additional parameters for the GetCapacity call.
-    StorageClassName string
-}
-
-// CSIStorageCapacityStatus contains the properties that can change over time.
-type CSIStorageCapacityStatus struct {
-    // AvailableCapacity is the value reported by the CSI driver in its GetCapacityResponse
-    // for a GetCapacityRequest with topology and parameters that match the
-    // CSIStorageCapacitySpec.
-    //
-    // The semantic is currently (CSI spec 1.2) defined as:
-    // The available capacity, in bytes, of the storage that can be used
-    // to provision volumes.
-    AvailableCapacity *resource.Quantity
+	metav1.TypeMeta
+	// Standard object's metadata. The name has no particular meaning. It must be
+	// be a DNS subdomain (dots allowed, 253 characters). To ensure that
+	// there are no conflicts with other CSI drivers on the cluster, the recommendation
+	// is to use csisc-<uuid>, a generated name, or a reverse-domain name which ends
+	// with the unique CSI driver name.
+	//
+	// Objects are namespaced.
+	//
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta
+
+	// NodeTopology defines which nodes have access to the storage
+	// for which capacity was reported. If not set, the storage is
+	// not accessible from any node in the cluster. If empty, the
+	// storage is accessible from all nodes.  This field is
+	// immutable.
+	//
+	// +optional
+	NodeTopology *metav1.LabelSelector
+
+	// The name of the StorageClass that the reported capacity applies to.
+	// It must meet the same requirements as the name of a StorageClass
+	// object (non-empty, DNS subdomain). If that object no longer exists,
+	// the CSIStorageCapacity object is obsolete and should be removed by its
+	// creator.
+	// This field is immutable.
+	StorageClassName string
+
+	// Capacity is the value reported by the CSI driver in its GetCapacityResponse
+	// for a GetCapacityRequest with topology and parameters that match the
+	// previous fields.
+	//
+	// The semantic is currently (CSI spec 1.2) defined as:
+	// The available capacity, in bytes, of the storage that can be used
+	// to provision volumes. If not set, that information is currently
```

Comment:
You should document here what does kube-scheduler do in the absence/presence of the field, and how it relates with maximumVolumeSize.

But... I think we can discuss this in the code PR.

Reply:
It's easier to discuss here because then I only need to change one file vs. many.

I added a section about behavior of kube-scheduler above for the entire CSIStorageCapacity struct. That seemed like a more natural place to discuss the relationship between different fields and how kube-scheduler handles an object. So you are saying that I should have that attached to each field?

Reply:
Ah, missed that. I think that is also part of the generated documentation in https://kubernetes.io/docs/reference/kubernetes-api/, so it should be fine.